aspnetcore/blazor/file-downloads.md

```diff
@@ -102,23 +100,31 @@ async function downloadFileFromStream(fileName, contentStreamReference) {
 function triggerFileDownload(fileName, url) {
   const anchorElement = document.createElement('a');
   anchorElement.href = url;
-
-  if (fileName) {
-    anchorElement.download = fileName;
-  }
-
+  anchorElement.download = fileName ?? '';
   anchorElement.click();
   anchorElement.remove();
 }
 ```
 
+`ArrayBuffer` effectively loads the entire file into client memory. For file downloads over 1 GB, we recommend opening a new tab for each large file download by setting the anchor's [`target`](https://developer.mozilla.org/docs/Web/HTML/Element/a#attr-target) to `_blank` and [`referrerpolicy`](https://developer.mozilla.org/docs/Web/HTML/Element/a#attr-referrerpolicy) to `no-referrer`:
+
+```javascript
+anchorElement.target = '_blank';
+anchorElement.referrerpolicy = 'no-referrer';
+```
```

Comment:
I think we may want to consider a more comprehensive example here. While these two lines open the download link in a new tab, we may want to show / mention how we rework the actual download to not use `contentStreamReference.arrayBuffer()`.

Maybe just creating a `triggerFileDownloadInNewTab` would do the trick.

Reply:
> rework the actual download to not use `contentStreamReference.arrayBuffer()` ... Maybe just creating a `triggerFileDownloadInNewTab` would do the trick.

Yes, could you provide the method for it? ... then I'll draft the text around it and ping u back.